src/frontend/src/app/assets/assets.component.ts

```diff
@@ -55,15 +90,15 @@ export class AssetsComponent implements OnInit {
   }
 
   public getAssetReading(): void {
-    console.log('Asset code: ', this.assetCode);
+    console.log('Asset code: ', this.asset['asset_code']);
     console.log('Limit: ', this.limit);
     console.log('offset: ', this.offset);
     
     this.assetsReadingsData = [];
-    if (this.assetCode.toLowerCase() === 'select') {
+    if (this.asset['asset_code'].toLowerCase() === 'select') {
```

Comment:
This if condition is not working yet. add a TODO comment to fix this.

Reply:
I have added TODO comment for the same in html file, will add that comment here.